### PR TITLE
Revert "Revert "docs(material/table): use declarative idiomatic RxJS in table-http-example (#22053)" (#22725)"

### PR DESCRIPTION
**Pull request by Splaktar**
- this reverts commit 9295ce5c
- add `detectChanges()` call to fix `NG0100: ExpressionChangedAfterItHasBeenCheckedError`

Brings back the changes from PR https://github.com/angular/components/pull/22053 and fixes the `ExpressionChangedAfterItHasBeenCheckedError` from https://github.com/angular/components/pull/22725.

### CODE REVIEW

File: src/components-examples/material/table/table-http/table-http-example.css [hunk 1]
```diff
@@ -6,7 +6,7 @@
 .example-table-container {
   position: relative;
   min-height: 200px;
-  max-height: 400px;
+  height: 400px;
```
Comment (Splaktar) [inline src/components-examples/material/table/table-http/table-http-example.css:9]: This is needed for the progress spinner to not overlap the input.

File: src/components-examples/material/table/table-http/table-http-example.ts [hunk 2]
```diff
@@ -57,7 +54,12 @@ export class TableHttpExample implements AfterViewInit {
           this.resultsLength = data.total_count;
           return data.items;
         })
-      ).subscribe(data => this.data = data);
+      );
+    this._changeDetectorRef.detectChanges();
```
Comment (crisbeto) [inline src/components-examples/material/table/table-http/table-http-example.ts:58]: Are we sure that this isn't masking the root cause of the error? Maybe the root cause is the `startWith` above which is updating the view immediately in `ngAfterViewInit` which is what usually causes "changed after checked" errors.

Comment (Splaktar) [reply src/components-examples/material/table/table-http/table-http-example.ts:58]: I can test that out some more, but I don't think that the error is due to the emission of the Observable, but instead the assignment of the Observable to the field, i.e. the field going from `undefined` to `Observable<GithubIssue[]>`.

Comment (andrewseguin) [reply src/components-examples/material/table/table-http/table-http-example.ts:58]: If we go this route we'd need some good commenting on why this is necessary. It doesn't immediately seem obvious that this is a better approach than having a `subscribe`

**Comment by andrewseguin** [comment]
Closing due to inactivity - feel free to re-open with a rebased version if you want us to continue the discussion on getting this in

**Pull request closed without merging**